Matrix.py

```diff
@@ -0,0 +1,175 @@
+        """divides a matrix by scalar
+                    args: self.data (tuple)- A matrix
+                        other(int/float) -A scalar
+
+                    returns: return the new matrix (tuple)
+
+                     raises: syntaxerror if "other" isn't a scalar"""
+        if type(other) is float or int:  # the program doesnt support division by other Matrix
+            return self.scalar_mul(1 / other)
+        else:
+            raise SyntaxError("only  scalar division")
+
+    def __mul__(self, other):
+        if type(other) is Matrix:
+            return self.matrices_mul(other)
+        elif type(other) is float or type(other) is int:
+            return self.scalar_mul(other)
+        else:
+            raise SyntaxError("invalid input")
+
+    def matrices_mul(self, other):
+        """multiply a matrix by matrix
+                    args: self.data (tuple)- A matrix
+                        other(tuple) -A matrix
+
+                    returns: return the new matrix (tuple)"""
+        if type(other) is Matrix:  # Matrix's multiplication --->
+            answer = []  # will contain the answer as a list of lists.
+            new_other = []
+            # this loop convert "other" from Matrix type to a list of lists named "new_other".
+            for line in range(len(self.data)):
+                new_other.append([])
+                answer.append([0] * len(self.data))  # ns starts as a list of lists full of zeros.
+                for num in range(len(self.data)):
+                    new_other[line].append(other[num][line])
+            # this is the multiplication process->
+            for tup in range(len(self.data)):
+                for num in range(len(self.data)):
+                    for i in range(len(self.data)):
+                        answer[tup][num] += self.data[tup][i] * new_other[num][i]
+#                here the program adds to the zero the summery of the multiplications of
+#                any two numbers which their places in the matrices are fits by the matrices
+#                multiplication rules.
+            sub_list = [tuple(ls) for ls in answer]
+            # convert answer to type Matrix. ls are the inside lists.
+            return Matrix((tuple(sub_list)))
+
+    def scalar_mul(self, other):
+        """multiply a matrix by a scalar
+                    args: self.data (tuple)- A matrix
+                        other(int/float) -A scalar
+
+                    returns: return the new matrix (tuple)
+
+                     raises: Syntaxerror if "other" isn't a scalar"""
+        answer = []
+        for tup in range(len(self.data)):
+            answer.append([])
+            for num in range(len(self.data)):
+                answer[tup].append(self.data[tup][num] * other)
+        sub_list = [tuple(ls) for ls in answer]
+        # convert ans to type Matrix. ls are the inside lists
+        return Matrix((tuple(sub_list)))
+
+    def __eq__(self, other):
```

Comment:
__neq__ missing

Reply:
it works for both cases

Reply:
Do you understand why it works for both cases?
eq and ne are not always depended, so implement ne